weave/flow/eval.py

```diff
@@ -237,21 +237,108 @@ async def predict_and_score(
+                            message = textwrap.dedent(
+                                f"""
+                                    You have created `{scorer_name}(column_map={scorer.column_map}, ...)`.
+
+                                    The `column_map` contains a key `{key}` which is not in the scorer's argument names.
+                                    Scorer argument names: {score_arg_names}
+
+                                    Hint:
+                                    - Ensure that the keys in `column_map` match the scorer's parameter names.
+                                    """
+                            )
+                            raise ValueError(message)
+
+                    for arg in score_arg_names:
+                        if arg == "output" or arg == "model_output":
+                            continue
+                        if arg in example:
+                            score_args[arg] = example[arg]
+                        elif arg in scorer.column_map:
+                            dataset_column_name = scorer.column_map[arg]
+                            if dataset_column_name in example:
+                                score_args[arg] = example[dataset_column_name]
+                            else:
+                                message = textwrap.dedent(
+                                    f"""
+                                        You have created `{scorer_name}(column_map={scorer.column_map}, ...)`.
+
+                                        You are mapping `{arg}` to `{dataset_column_name}`, but `{dataset_column_name}`
+                                        was not found in the dataset columns.
+                                        
+                                        Available dataset columns: {list(example.keys())}
+
+                                        Hint:
+                                        - Ensure that `column_map` maps scorer parameter names to existing dataset column names.
+                                        """
+                                )
+                                raise ValueError(message)
+                        else:
+                            message = textwrap.dedent(
+                                f"""
+                                    You have created `{scorer_name}(column_map={scorer.column_map}, ...)`.
+
+                                    Scorer argument `{arg}` is not found in the dataset columns and is not mapped in `column_map`.
+                                    
+                                    Available dataset columns: {list(example.keys())}
+                                    `column_map`: {scorer.column_map}
+
+                                    Hint:
+                                    - Either provide `{arg}` directly in the dataset, or map it via `column_map`.
+                                    """
+                            )
+                            raise ValueError(message)
```

Comment:
These errors are helpful, but it might be annoying to raise them 1-by-1.  Can collect all the errors and raise them at once?

Reply:
You mean do all the checks at once? I think we'd prefer to break it up like this for now